### ISSUE TITLE
No Backend Found When Using File Engine with CN

### ISSUE THREAD

**Issue by sharon-clue** [StarRocks/starrocks#34514]
I'm running a StarRocks cluster on AWS EKS with `shared_data` mode.
The cluster has one CN and 3 FEs.
The deployment was done with the latest Helm Charts (`v1.8.5`)

### Steps to reproduce the behavior
```
SELECT * FROM FILES(
    "path" = "s3://clue-data-dev-airflow-staging/starrocks-test/parquet-files/mt_cars.parquet",
    "format" = "parquet",
    "enable_recursive_listing" = "true",
	"aws.s3.use_aws_sdk_default_behavior" = "true",
    "aws.s3.enable_ssl" = "true",
    "aws.s3.region" = "eu-west-1",
    "aws.s3.enable_path_style_access" = "true"
) LIMIT 10;
```

### Expected behavior (Required)
Return the result of the query

### Real behavior (Required)
`SQL Error [1064] [42000]: Access storage error. Error message: Failed to send proxy request. No alive backends`

### StarRocks version (Required)
`3.1.4-0c4b2a3`

Additionally, what I noticed was that when I manually assigned the CN as a BE the query from files did work, but then other queries (anything running directly on StarRocks) didn't were stuck.
I contacted the SR guys on Slack and after explaining the situation we agreed that it's a bug.

**Comment by Jin-H** [comment]
probably the same as this https://github.com/StarRocks/starrocks/pull/34270

**Comment by Jin-H** [comment]
@sharon-clue could you provide more detailed reproduction process?

**Comment by sharon-clue** [comment]
**Steps to reproduce the behavior:**

1. deploy starrocks in shared_data mode
2. run the following command

**Comment by Jin-H** [comment]
ok

**Comment by sharon-clue** [comment]
i tried this on 3.2 and still get the same error.
would it be possible to fix this?

**Comment by alberttwong** [comment]
Did this get resolved?

**Comment by Jin-H** [comment]
> Did this get resolved?

Yes.

![image](https://github.com/StarRocks/starrocks/assets/16933180/1681d269-1320-46f4-9914-b651e7b3ac3f)

**Comment by Jin-H** [comment]
> i tried this on 3.2 and still get the same error. would it be possible to fix this?



> i tried this on 3.2 and still get the same error. would it be possible to fix this?

I think the StarRocks you tested is version 3.2-RC1, which was fixed in the 3.2 release version.

**Comment by sharon-clue** [comment]
i tried it on 3.1.5 and it works.
thank you!!
